oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/pkce/endpoint/Sha256CodeChallengeTransformation.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core.pkce.endpoint;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+/**
+ * A @{@link CodeChallengeTransformation} for the transformation of Code Verifier to
+ * a Code Challenge using SHA-256.
+ *
+ * @author Stephen Doxsee
+ * @author Kevin Bolduc
+ * @since 5.2
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7636#section-4.2">4.2 Client Creates the Code Challenge</a>
+ */
+public class Sha256CodeChallengeTransformation implements CodeChallengeTransformation {
```

Comment:
Please remove this and encapsulate the `code_challenge` computation in `DefaultOAuth2AuthorizationRequestResolver`. We can expose later if need be.